security/middleware.py

```diff
@@ -410,49 +439,63 @@ class ContentSecurityPolicyMiddleware:
     def _csp_builder(self, csp_dict):
         csp_string = ""
 
-        for k,v in csp_dict.items():
+        for k, v in csp_dict.items():
 
+            # check if the directive is on the list of those requiring an array argument
             if k in self._CSP_LOC_TYPES:
 
-                if not type(v) == list:
-                    logger.warning('Arguments to {0} must be given as array'.format(k))
+                if type(v) is not list:
+                    logger.warning('Arguments to {} must be given as an array, found {} instead ({}) - ignoring'.format(k, type(v), v))
```

Comment:
Sorry, I updated my comment, they work in 2.7 for sure, but they don't work in anything prior to that version. (So they fail in 2.6 for example, which is what we still use on some of our servers..)